test/lint/lint-circular-dependencies.sh

```diff
@@ -64,6 +64,7 @@ EXPECTED_CIRCULAR_DEPENDENCIES=(
     "logging -> util/system -> stacktraces -> logging"
     "logging -> util/system -> util/getuniquepath -> random -> logging"
     "coinjoin/client -> coinjoin/util -> wallet/wallet -> coinjoin/client"
+    "coinjoin/client -> coinjoin/util -> wallet/wallet -> psbt -> node/transaction -> node/context -> coinjoin/context -> coinjoin/client"
```

Comment:
note for further refactoring: `coinjoin/util` should not depends on wallet/wallet! It can depends only on `interface/wallet`. (may be out-of-scope of current PR)